spring-boot-3-integrationtests/src/test/java/org/axonframework/eventhandling/deadletter/DeadLetteringEventIntegrationTest.java

```diff
@@ -0,0 +1,726 @@
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.axonframework.common.transaction.NoOpTransactionManager;
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.eventhandling.EventHandler;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.StreamingEventProcessor;
+import org.axonframework.eventhandling.pooled.PooledStreamingEventProcessor;
+import org.axonframework.eventhandling.tokenstore.inmemory.InMemoryTokenStore;
+import org.axonframework.messaging.MetaData;
+import org.axonframework.messaging.annotation.MessageIdentifier;
+import org.axonframework.messaging.deadletter.DeadLetter;
+import org.axonframework.messaging.deadletter.Decisions;
+import org.axonframework.messaging.deadletter.EnqueuePolicy;
+import org.axonframework.messaging.deadletter.SequencedDeadLetterQueue;
+import org.axonframework.messaging.unitofwork.RollbackConfigurationType;
+import org.axonframework.utils.InMemoryStreamableEventSource;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.function.Predicate;
+
+import static org.awaitility.Awaitility.await;
+import static org.axonframework.eventhandling.GenericEventMessage.asEventMessage;
+import static org.axonframework.springboot.utils.AssertUtils.assertWithin;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test class validating the combination of an {@link org.axonframework.eventhandling.EventProcessor} containing a
+ * {@link DeadLetteringEventHandlerInvoker} (a specific type of Processing Group). This test validates that:
+ *
+ * <ul>
+ *     <li>Handled {@link EventMessage EventMessages} are enqueued in a {@link SequencedDeadLetterQueue}
+ *     if event handling fails.</li>
+ *     <li>Handled {@code EventMessage EventMessages} are enqueued in a {@code DeadLetterQueue}
+ *     if a previous event in that sequence was enqueued.</li>
+ *     <li>Enqueued {@code EventMessage EventMessages} are successfully evaluated and evicted
+ *     from a {@code DeadLetterQueue} on {@link DeadLetteringEventHandlerInvoker#process(Predicate)}}.</li>
+ *     <li>Enqueued {@code EventMessage EventMessages} are unsuccessfully evaluated
+ *     and requeued in the {@code DeadLetterQueue} on {@link DeadLetteringEventHandlerInvoker#process(Predicate)}.</li>
+ *     <li>Concurrently publish events that succeed and/or fail and validate the evaluation
+ *     by the {@code DeadLetterQueue}.</li>
+ *     <li>Concurrently publish events in bulk that succeed and/or fail and validate the evaluation
+ *     by the {@code DeadLetterQueue}.</li>
+ * </ul>
+ *
+ * @author Steven van Beelen
+ */
+public abstract class DeadLetteringEventIntegrationTest {
```

Comment:
I think you should be able to extract the `DeadLetteringEventIntegrationTest` as a test dependency instead of copying it.
The `axon-spring` module does that too:
```xml
<dependency>
    <groupId>org.axonframework</groupId>
    <artifactId>axon-messaging</artifactId>
    <version>${project.version}</version>
    <type>test-jar</type>
    <scope>test</scope>
</dependency>
```